sdk/src/types/block/output/basic.rs

```diff
@@ -152,8 +156,34 @@ impl BasicOutputBuilder {
         self
     }
 
+    /// Adds a storage deposit if one is needed to cover the current amount.
+    pub fn with_sufficient_storage_deposit(
+        self,
+        return_address: impl Into<Address>,
+        rent_structure: RentStructure,
+        token_supply: u64,
+    ) -> Result<Self, Error> {
+        Ok(match self.amount {
+            OutputBuilderAmount::Amount(amount) => {
+                let rent_cost = self.rent_cost(rent_structure);
+                if amount < rent_cost {
+                    self.with_amount(amount + rent_cost).replace_unlock_condition(
+                        StorageDepositReturnUnlockCondition::new(return_address, rent_cost, token_supply)?,
+                    )
```

Comment:
But it's adding if there is none yet

Reply:
If the amount is not 0 it would add more than actually required
`amount + rent_cost`, because to return something we just need the minimum required storage deposit of an output, so the difference between that and the rent_cost could be subtracted from the `amount` part of `amount + rent_cost` (if `amount` would then be negative, this number would need to be added to the return amount)

So for example provided amount: 10
`rent_cost_basic` =  min required storage deposit basic output: 42600
`rent_cost_basic_sdr` = min required storage deposit basic output with storage deposit return: 42800

Then `amount + rent_cost_basic_sdr` == 10+42800 => 42810
But 42800 would be sufficient, because we can just have have 42790 in the storage deposit return unlock condition

Reply:
Yeah but you're trying to send 100 to someone and this would change that

Reply:
How does this change? The recipient would still get exactly the same amount

Reply:
Oh, I misread. You are correct.